lib/services/unified_resource_adapter.go

```diff
@@ -0,0 +1,155 @@
+package services
+
+import (
+	"encoding/json"
+	"time"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils"
+)
+
+// UnifiedResource153 is a type constraint that requires a type to implement
+// the Resource153 interface AND provides a Clone method.
+type UnifiedResource153[T interface{ CloneResource() T }] interface {
+	types.Resource153
+	CloneResource() T
+}
```

Comment:
Why is this needed? Is there a way to avoid another templated interface type? Can't types.Resource153 just include Clone method?

Reply:
@r0mant, @rosstimothy: It will also need `ResourceWithLabels`, but sure. 

The whole `UnifiedResource153Adapter` exists because I was trying to get a RFD153-style Resource into the Unified Resource Cache while still following follow the advice in  the [types.Resource153 godoc](https://github.com/gravitational/teleport/blob/72825b1b9e1cc47a5c5468a910258405e4b238b3/api/types/resource_153.go#L32), specifically:

> `Embedding or further extending this interface is highly discouraged.`

... and this was a way to 

- avoid expanding an interface when explicitly advised not to (the only place `Resource153` is mentioned is the type constraint on `UnifiedResource153`),
- avoid inadvertently changing the behaviour of existing values wrapped by `resource153ToLegacyAdapter`, and
- stop any attempt to wrap a resource type without a `Clone()` method at compile time. 

My gut feeling is that the "correct" thing to do is probably slim down the [`resource`](https://github.com/gravitational/teleport/blob/72825b1b9e1cc47a5c5468a910258405e4b238b3/lib/services/unified_resource.go#L723) interface used in the Resource Cache as per the `Resource153` godoc: 

> If you do need to represent resources in a generic manner, consider
> declaring a smaller interface with only what you need.

But that refactor is well outside the scope of this PR. 

Anyway, that was my reasoning for things being as they are. If expanding the `types.Resource153` interface (and the accompanying `resource153ToLegacyAdapter` etc) is the lesser of two evils, I'm happy to go that way.